moveit_ros/moveit_servo/test/basic_servo_tests.cpp

```diff
@@ -32,154 +32,134 @@
-
-    // Create moveit_servo
-    servo_ = std::make_shared<Servo>(nh_, planning_scene_monitor_);
-  }
-  void TearDown() override
-  {
-  }
-
-protected:
-  ros::NodeHandle nh_{ "~" };
-  planning_scene_monitor::PlanningSceneMonitorPtr planning_scene_monitor_;
-  moveit_servo::ServoPtr servo_;
-};  // class ServoFixture
-
 TEST_F(ServoFixture, SendTwistStampedTest)
 {
-  servo_->start();
+  auto log_time_start = node_->now();
+  ASSERT_TRUE(setupStartClient());
+  auto log_time_end = node_->now();
+  RCLCPP_INFO_STREAM(LOGGER, "Setup time: " << (log_time_end - log_time_start).seconds());
+
+  // Start Servo
+  ASSERT_TRUE(start());
+  EXPECT_EQ(latest_status_, moveit_servo::StatusCode::NO_WARNING);
 
-  auto parameters = servo_->getParameters();
+  auto parameters = servo_parameters_;
 
   // count trajectory messages sent by servo
   size_t received_count = 0;
-  boost::function<void(const trajectory_msgs::JointTrajectoryConstPtr&)> traj_callback =
-      [&received_count](const trajectory_msgs::JointTrajectoryConstPtr& /*msg*/) { ++received_count; };
-  auto traj_sub = nh_.subscribe(parameters.command_out_topic, 1, traj_callback);
+  std::function<void(const trajectory_msgs::msg::JointTrajectory::ConstSharedPtr)> traj_callback =
+      [&received_count](const trajectory_msgs::msg::JointTrajectory::ConstSharedPtr msg) { ++received_count; };
+  auto traj_sub = node_->create_subscription<trajectory_msgs::msg::JointTrajectory>(
+      resolveServoTopicName(parameters->command_out_topic), 1, traj_callback);
 
   // Create publisher to send servo commands
-  auto twist_stamped_pub = nh_.advertise<geometry_msgs::TwistStamped>(parameters.cartesian_command_in_topic, 1);
+  auto twist_stamped_pub = node_->create_publisher<geometry_msgs::msg::TwistStamped>(
+      resolveServoTopicName(parameters->cartesian_command_in_topic), 1);
 
   constexpr double test_duration = 1.0;
-  const double publish_period = parameters.publish_period;
+  const double publish_period = parameters->publish_period;
   const size_t num_commands = static_cast<size_t>(test_duration / publish_period);
 
   // Set the rate differently from the publish period from the parameters to show that
   // the number of outputs is set by the number of commands sent and not the rate they are sent.
-  ros::Rate publish_rate(2. / publish_period);
+  rclcpp::Rate publish_rate(2. / publish_period);
 
   // Send a few Cartesian velocity commands
-  for (size_t i = 0; i < num_commands && ros::ok(); ++i)
+  for (size_t i = 0; i < num_commands && rclcpp::ok(); ++i)
   {
-    auto msg = moveit::util::make_shared_from_pool<geometry_msgs::TwistStamped>();
-    msg->header.stamp = ros::Time::now();
+    auto msg = moveit::util::make_shared_from_pool<geometry_msgs::msg::TwistStamped>();
+    msg->header.stamp = node_->now();
     msg->header.frame_id = "panda_link0";
     msg->twist.angular.y = 1.0;
 
     // Send the message
-    twist_stamped_pub.publish(msg);
+    twist_stamped_pub->publish(*msg);
```

Comment:
De-referencing the message here will cause an unnecessary copy.

```suggestion
    twist_stamped_pub->publish(msg);
```

Reply:
This would not compile as I declared msg from `moveit::util::make_shared_from_pool<control_msgs::msg::JointJog>();` which is a sharedPtr. However I can declare it as `control_msgs::msg::JointJog msg;` and do the same thing.

Reply:
Sending the new commit right now with this change. Appreciate the review, thanks!

Reply:
That is because the approach of `moveit::util::make_shared_from_pool` won't work for ros2 because the publish interface only takes either the message type itself or a unique_ptr to the message type.  You can try changing `make_shared_from_pool` to create `unique_ptr`.  Here is the documentation on it: https://docs.ros.org/en/foxy/Tutorials/Intra-Process-Communication.html

I'm sorry I didn't think of this before.  It might also might be worth removing `moveit::util::make_shared_from_pool` from the ros2 version of this until we figure out and benchmark if it is worthwhile to generate unique_ptr with this and just use `std::make_unique` for now.

Reply:
Learned something new today, thanks for the clarrification. Should I open a issue / PR to remove / uncomment `moveit::util::make_shared_from_pool`? Or I can open one up for benchmarking it to unique_ptr?

Reply:
Seems like we would probably want to just delete it. I'd open an issue for that.

Benchmarking would be interesting if you have tons of free time    ;)